### PR TITLE
Add vision native

### PR DESCRIPTION
**Pull request by nikwalia**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -188,15 +220,15 @@ lazy val docsMappingsAPIDir = settingKey[String]("Name of subdirectory in site t
 lazy val docs = project
   .enablePlugins(ScalaUnidocPlugin)
   .dependsOn(coreJVM, modelJVM, controlJVM,
-    remote, vision, frcJVM, configJVM, sensorsJVM,
+    remote, visionJVM, visionNative, frcJVM, configJVM, sensorsJVM,
```
Comment (shadaj) [inline build.sbt]: Only `visionJVM` here.

File: build.sbt [hunk 2]
```diff
@@ -188,15 +220,15 @@ lazy val docsMappingsAPIDir = settingKey[String]("Name of subdirectory in site t
...
     commonsJVM, lighting)
   .settings(
     autoAPIMappings := true,
     docsMappingsAPIDir := "api",
     addMappingsToSiteDir(mappings in (ScalaUnidoc, packageDoc), docsMappingsAPIDir),
     unidocProjectFilter in (ScalaUnidoc, unidoc) :=
       inProjects(coreJVM, modelJVM, controlJVM,
-        remote, vision, frcJVM, configJVM, lighting, sensorsJVM, commonsJVM)
+        remote, visionJVM, visionNative, frcJVM, configJVM, lighting, sensorsJVM, commonsJVM)
```
Comment (shadaj) [inline build.sbt]: Same as above.

File: build.sbt [hunk 3]
```diff
@@ -100,12 +101,43 @@ lazy val remote = project.dependsOn(coreJVM).settings(
...
+  resolvers += "WPILib-Maven" at "http://team846.github.io/wpilib-maven",
+  resolvers += "opencv-maven" at "https://github.com/WPIRoboticsProjects/opencv-maven/raw/mvn-repo",
+
+  libraryDependencies += "edu.wpi.first" % "cscore" % "2018.1.1",
+  libraryDependencies += "org.opencv" % "opencv-java" % "3.1.0",
+  libraryDependencies += "edu.wpi.first" % "ntcore" % wpiVersion
+).nativeSettings(
+  if (System.getenv("NATIVE_TARGET") == "ARM32") {
+    Seq(
+      libraryDependencies += "com.lynbrookrobotics" %%% "wpilib-scala-native" % "0.1-SNAPSHOT",
```
Comment (shadaj) [inline build.sbt:115]: You're using a bunch of different versions here. Can you extract out the native version into a variable?

File: build.sbt [hunk 4]
```diff
@@ -100,12 +101,43 @@ lazy val remote = project.dependsOn(coreJVM).settings(
...
+    )
+  } else Seq(
+    resolvers += "WPILib-Maven" at "http://team846.github.io/wpilib-maven",
+
+    libraryDependencies += "org.opencv" % "opencv-java" % "3.1.0",
+    libraryDependencies += "edu.wpi.first" % "cscore" % "2018.1.1",
+    libraryDependencies += "edu.wpi.first" % "wpilib" % wpiVersion,
+    libraryDependencies += "edu.wpi.first" % "ntcore" % wpiVersion,
+    libraryDependencies += "com.ctre" % "phoenix" % ctreVersion,
+    libraryDependencies += "com.lynbrookrobotics" % "ntcore-scala-native" % "0.1.0+4-f5af12c9",
```
Comment (shadaj) [inline build.sbt]: This dependency is only available for ARM32. It should be removed.

File: build.sbt [hunk 5]
```diff
@@ -100,12 +101,43 @@ lazy val remote = project.dependsOn(coreJVM).settings(
...
+      libraryDependencies += "com.lynbrookrobotics" %%% "wpilib-scala-native" % "0.1-SNAPSHOT",
+      libraryDependencies += "edu.wpi.first" % "cscore" % "2018.1.1",
+      libraryDependencies += "org.opencv" % "opencv-java" % "3.1.0",
+      libraryDependencies += "com.lynbrookrobotics" %%% "ntcore-scala-native" % "0.1.0+4-f5af12c9",
+      libraryDependencies += "com.lynbrookrobotics" %%% "phoenix-scala-native" % "0.1-SNAPSHOT"
+    )
+  } else Seq(
+    resolvers += "WPILib-Maven" at "http://team846.github.io/wpilib-maven",
+
+    libraryDependencies += "org.opencv" % "opencv-java" % "3.1.0",
```
Comment (shadaj) [inline build.sbt]: You should only depend on what you need. So the only dependency should be ntcore.

File: build.sbt [hunk 6]
```diff
@@ -100,12 +101,43 @@ lazy val remote = project.dependsOn(coreJVM).settings(
...
+  resolvers += "opencv-maven" at "https://github.com/WPIRoboticsProjects/opencv-maven/raw/mvn-repo",
+
+  libraryDependencies += "edu.wpi.first" % "cscore" % "2018.1.1",
+  libraryDependencies += "org.opencv" % "opencv-java" % "3.1.0",
+  libraryDependencies += "edu.wpi.first" % "ntcore" % wpiVersion
+).nativeSettings(
+  if (System.getenv("NATIVE_TARGET") == "ARM32") {
+    Seq(
+      libraryDependencies += "com.lynbrookrobotics" %%% "wpilib-scala-native" % "0.1-SNAPSHOT",
+      libraryDependencies += "edu.wpi.first" % "cscore" % "2018.1.1",
```
Comment (shadaj) [inline build.sbt]: You should only be depending on ntcore here.

**Review by shadaj — DISMISSED** [review]
First pass

**Review by PhilipAxelrod — APPROVED** [review]
lgtm

**Pull request merged**